pkg/frontend/openshiftcluster_preflightvalidation_test.go

```diff
@@ -6,96 +6,342 @@ package frontend
+			},
+		},
+		{
+			name: "Failed Preflight Static Invalid ResourceGroup",
+			fixture: func(f *testdatabase.Fixture) {
+				f.AddSubscriptionDocuments(api.ExampleSubscriptionDocument())
+			},
+			preflightRequest: func() *api.PreflightRequest {
+				return &api.PreflightRequest{
+					Resources: []json.RawMessage{
+						[]byte(fmt.Sprintf(`
+							{
+								"apiVersion": "%s",
+								"id": "%s",
+								"name": "%s",
+								"type": "%s",
+								"location": "%s",
+								"properties": {
+									"clusterProfile": {
+										"domain": "%s",
+										"resourceGroupId": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/resourcenameTest",
+										"fipsValidatedModules": "%s"
+									}
+								}
+							}
+							`, apiVersion, clusterId,
+							api.ExampleOpenShiftClusterDocument().OpenShiftCluster.Name,
+							api.ExampleOpenShiftClusterDocument().OpenShiftCluster.Type,
+							location, defaultProfile, api.FipsValidatedModulesEnabled)),
+					},
+				}
+			},
+			wantStatusCode: http.StatusOK,
+			wantResponse: &api.ValidationResult{
+				Status: api.ValidationStatusFailed,
+				Error: &api.CloudErrorBody{
+					Message: "400: InvalidParameter: properties.clusterProfile.resourceGroupId: The provided resource group '/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/resourcenameTest' is invalid: must be in same subscription as cluster.",
+				},
+			},
+		},
+		{
+			name: "Failed Preflight Cluster Check",
+			fixture: func(f *testdatabase.Fixture) {
+				f.AddSubscriptionDocuments(api.ExampleSubscriptionDocument())
+			},
+			preflightRequest: func() *api.PreflightRequest {
+				return &api.PreflightRequest{
+					Resources: []json.RawMessage{
+						[]byte(fmt.Sprintf(`
+							{
+								"apiVersion": "%s",
+								"id": "%s",
+								"name": "%s",
+								"type": "%s",
+								"location": "%s",
+								"properties": {}
+							}
+							`, apiVersion, "resourceGroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName",
+							api.ExampleOpenShiftClusterDocument().OpenShiftCluster.Name,
+							api.ExampleOpenShiftClusterDocument().OpenShiftCluster.Type,
+							location)),
+					},
+				}
+			},
+			wantStatusCode: http.StatusOK,
+			wantResponse: &api.ValidationResult{
+				Status: api.ValidationStatusFailed,
+				Error: &api.CloudErrorBody{
+					Message: "400: Cluster not found for resourceID: resourcegroups/resourcegroup/providers/microsoft.redhatopenshift/openshiftclusters/resourcename",
+				},
+			},
+		},
+		{
+			name: "Failed Preflight Install Version",
+			fixture: func(f *testdatabase.Fixture) {
+				f.AddSubscriptionDocuments(api.ExampleSubscriptionDocument())
+			},
+			changeFeed: defaultVersionChangeFeed,
+			preflightRequest: func() *api.PreflightRequest {
+				return &api.PreflightRequest{
+					Resources: []json.RawMessage{
+						[]byte(fmt.Sprintf(`
+							{
+								"apiVersion": "%s",
```

Comment:
nit: same as above - now thinking this should be wrapped in a function call, where you pass it a map of variables.